### ISSUE TITLE
Documentation Request: Sample Project

### ISSUE THREAD

**Issue by divergio** [jcampbell05/xcake#118]
More than a sample `Cakefile`, I think a working sample project would be very illuminating for beginner Xcake users. 

I'm trying to convert a project now, and although Xcake will `make` just fine, the resulting project file can't be parsed by Xcode. I assume this is due to an error in the `Cakefile`. 

Of course, more **complete** `Cakefile` validation would help solve this issue as well, but I think it would also be helpful to have a working **sample project** to compare to.

A working sample project would help new users to see both the configuration and the results. Because the current sample `Cakefile` is just a fake `Cakefile` and not a whole sample project, I can't actually build it and see the output. As a result, I can't do the "sanity check" that my basic configuration is correct (and that Xcake is compatible with my current version of Xcode).

**Comment by jcampbell05** [comment]
@divergio Thanks for raising this, I'll try to build a better example and get it working with all Xcode versions.

**Comment by maximkhatskevich** [comment]
@divergio to be honest I think that having a whole sample project just does not make sense. The sample Cakefile(s) are pretty self-explanatory. This is the idea behind Xcake - you write easy to read declarative configuration file and Xcake takes care about generating whole Xcode project based on this.

After learning the sample files, If you do not understand what's going there - probably you need to learn more about how Xcode project file is organized first. Also, in my opinion, Xcake is not supposed to be used by beginners and/or for small projects with single developer. It's a tool for saving time for mid- and senior level developers  when they work on a relatively big project with multiple collaborators. So probably you just don't need this tool yet? @jcampbell05 correct me if I'm wrong.

If you are still sure it's a tool you really need in your development process and you are trying to write Cakefile for your project, but project generated from your Cakefile does not work properly - try to comment almost everything in your Cakefile and keep there only very basic configurations. Generate the project and check if it works. The ideas is to make a very-very basic Cakefile first, make sure generated project opens properly by Xcode (even if it's empty yet). After you find minimum project configuration that generates functional project file - start uncommenting more configurations, line by line. After each addition - remove existing project file and regenerate project. At some point you'll find which exact line leads to a broken project file, that can not be opened by Xcode properly. At this point you most likely will get the idea what's wrong and how to fix it. Or you can come back here and ask us.

Hope that will help you resolve your issue. Cheers!

**Comment by divergio** [comment]
@maximkhatskevich Thanks for your reply. 

Perhaps I wasn't clear, I'm not referring to beginner Xcode users, I'm referring to people trying out xcake for the first time or perhaps just evaluating xcake for use in their project. The relatively large project with multiple collaborators that I'm working on is exactly why I am interested in xcake (I'm **so** sick of digging through `pbxproj` files to resolve merge errors!). 

I already resolved my issue getting a Cakefile to work using a trial-and-error approach similar to what you suggested. The reason I suggested a sample project is to help others avoid this step, or at least to have one working version they can play with and take apart to learn how xcake works. 

Further, I think it's best to figure out how a Cakefile works and to test its functionality for your own project by trying it out on a smaller project first (i.e. don't jump straight into trying to convert your 1000 file project). A sample project provides a place for new/potential users to do this test. 

To reiterate, the drawback of providing only a sample Cakefile is that it can't actually generate a working project file. The files it references aren’t there. It is documentation, not a demonstration.
